src/main/java/org/openrewrite/staticanalysis/NeedBraces.java

```diff
@@ -111,6 +133,11 @@ private <T extends Statement> J.Block buildBlock(Statement owner, T element) {
         @Override
         public J.Block visitBlock(J.Block block, ExecutionContext ctx) {
             J.Block bl = super.visitBlock(block, ctx);
+            Boolean removeEndComments = getCursor().pollMessage("removeEndComments");
+            if (removeEndComments != null) {
```

Comment:
Should we not use the `Boolean.TRUE.equals(Object)` to not have a classCastException when this message would ever be incorrectly set to a String/int....
Also, if the message would be set to false, the if block is still executing in the as-is implementation. 

```suggestion
            if (Boolean.TRUE.equals(getCursor().pollMessage("removeEndComments"))) {
```